### PR TITLE
Direct formatted output from `as_result_df`

### LINKED ISSUE #784: Is there a way to get a data frame of the final TableTree object?
### What is your question?

If I have a TableTree object `tbl`, is there a way to make the main table what was created (what I see with `Viewer(tbl)`) into a data frame or a data.table? I know there is `as_result_df()` but this function returns the raw values and not the final calculated values. I know the columns will likely be characters, which is fine but are there other options besides `as_result_df()`?.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by Melkiades**
addition of `as_strings`, `as_viewer`, and `expand_colnames` for what I imagine is easier QCing. @kieranjmartin fyi

@mrmorgan17 try it out. Options are described in `?as_result_df`

Fixes #784

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -34,13 +34,13 @@ Depends:
     methods,
     R (>= 2.10)
 Imports:
+    checkmate (>= 2.1.0),
```
Comment (Melkiades) [inline DESCRIPTION:37]: I think it is time @edelarua @shajoezhu

File: R/tt_export.R [hunk 2]
```diff
@@ -53,52 +49,254 @@ import_from_tsv <- function(file) {
...
+    )
+  )
+  
+  ret <- cbind(
+    metadf[metadf$node_class != "LabelRow", ],
+    cellvals
+  )
+  
+  # If we want to expand colnames
+  if (expand_colnames) {
```
Comment (Melkiades) [inline R/tt_export.R:247]: and here

File: tests/testthat/test-result_data_frame.R [hunk 3]
```diff
@@ -67,3 +66,36 @@ test_that("Result Data Frame generation works v0", {
...
+  string_tbl$row_name[seq_len(mf_nlheader(mf))] <- res$row_name[seq_len(mf_nlheader(mf))]
+  expect_equal(res, string_tbl)
+  
+  expect_silent(basic_table() %>% build_table(DM) %>% as_result_df())
+  tbl <- basic_table() %>% analyze("BMRKR1") %>% build_table(DM) 
+  
+  expect_equal(as_result_df(tbl)$V1, 5.851948, tolerance = 1e-6) # V1?
+
+  # as_result_df(tbl, as_strings = TRUE)  
+  # as_result_df(tbl, as_viewer = TRUE)
```
Comment (Melkiades) [inline tests/testthat/test-result_data_frame.R:99]: Still to fix

**Comment by edelarua** [comment]
Hi @Melkiades,

Looks great so far! Thanks for working on this as I think it's much needed :)

One thing I noticed - when there are mixed numeric types (i.e. integers & doubles) in a single column, the integers are converted to NAs. I know you can't mix types in a data frame column, so would it be possible to instead convert the integers to doubles, which I feel would be more practical? It works fine if `as_strings = TRUE` but I would assume this would often be useful in the `as_strings = FALSE` case as well to extract values.

Example:

```r
lyt <- basic_table() %>%
    split_cols_by("ARM") %>%
    split_rows_by("STRATA1") %>%
    analyze(c("AGE", "BMRKR2"))

tbl <- build_table(lyt, ex_adsl)
as_result_df(tbl, simplify = TRUE, as_viewer = TRUE)
```

```
   row_name A: Drug X B: Placebo C: Combination
1      Mean     33.08      35.11          34.23
2       LOW        NA         NA             NA
3    MEDIUM        NA         NA             NA
4      HIGH        NA         NA             NA
5      Mean     33.85         36          36.33
6       LOW        NA         NA             NA
7    MEDIUM        NA         NA             NA
8      HIGH        NA         NA             NA
9      Mean     34.22      35.18          35.63
10      LOW        NA         NA             NA
11   MEDIUM        NA         NA             NA
12     HIGH        NA         NA             NA
```

**Comment by Melkiades** [comment]
Thanks, @edelarua, for finding this problem at the source! It should be fixed now ;)

**Comment by edelarua** [comment]
> Thanks, @edelarua, for finding this problem at the source! It should be fixed now ;)

Perfect! It's all working for me now. You may want to wait for feedback from others but if not I think it's good to go :)

**Review by edelarua — APPROVED** [review]

**Pull request merged**